primitives/arkworks/src/utils.rs

```diff
@@ -1,124 +1,150 @@
 
-	serialize_result(result)
+	let result: ArkScale<twisted_edwards::Projective<Curve>> = result.into();
+	Ok(<ArkScale<twisted_edwards::Projective<Curve>> as Encode>::encode(&result))
 }
 
-/// Compute a projective scalar multiplication on G2 through arkworks
-pub fn mul_projective_generic<Group: SWCurveConfig>(base: Vec<u8>, scalar: Vec<u8>) -> Vec<u8> {
-	let base = deserialize_argument::<short_weierstrass::Projective<Group>>(&base);
-	let scalar = deserialize_argument::<Vec<u64>>(&scalar);
+pub(crate) fn mul_projective_generic<Group: SWCurveConfig>(
+	base: Vec<u8>,
+	scalar: Vec<u8>,
+) -> Result<Vec<u8>, ()> {
+	let base = <ArkScale<short_weierstrass::Projective<Group>> as Decode>::decode(
+		&mut base.clone().as_slice(),
+	)
+	.map_err(|_| ())?;
+	let scalar =
+		<ArkScale<Vec<u64>> as Decode>::decode(&mut scalar.clone().as_slice()).map_err(|_| ())?;
 
-	let result = <Group as SWCurveConfig>::mul_projective(&base, &scalar);
+	let result = <Group as SWCurveConfig>::mul_projective(&base.0, &scalar.0);
 
-	serialize_result(result)
+	let result: ArkScale<short_weierstrass::Projective<Group>> = result.into();
+	Ok(<ArkScale<short_weierstrass::Projective<Group>> as Encode>::encode(&result))
 }
 
-/// Compute a projective scalar multiplication for twisted_edwards through arkworks
-pub fn mul_projective_te_generic<Group: TECurveConfig>(base: Vec<u8>, scalar: Vec<u8>) -> Vec<u8> {
-	let base = deserialize_argument::<twisted_edwards::Projective<Group>>(&base);
-	let scalar = deserialize_argument::<Vec<u64>>(&scalar);
+pub(crate) fn mul_projective_te_generic<Group: TECurveConfig>(
+	base: Vec<u8>,
+	scalar: Vec<u8>,
+) -> Result<Vec<u8>, ()> {
+	let base = <ArkScale<twisted_edwards::Projective<Group>> as Decode>::decode(
+		&mut base.clone().as_slice(),
+	)
+	.map_err(|_| ())?;
+	let scalar =
+		<ArkScale<Vec<u64>> as Decode>::decode(&mut scalar.clone().as_slice()).map_err(|_| ())?;
 
-	let result = <Group as TECurveConfig>::mul_projective(&base, &scalar);
+	let result = <Group as TECurveConfig>::mul_projective(&base.0, &scalar.0);
 
-	serialize_result(result)
+	let result: ArkScale<twisted_edwards::Projective<Group>> = result.into();
+	Ok(<ArkScale<twisted_edwards::Projective<Group>> as Encode>::encode(&result))
 }
 
-/// Compute a affine scalar multiplication through arkworks
-pub fn mul_affine_generic<Group: SWCurveConfig>(base: Vec<u8>, scalar: Vec<u8>) -> Vec<u8> {
-	let base = deserialize_argument::<short_weierstrass::Affine<Group>>(&base);
-	let scalar = deserialize_argument::<Vec<u64>>(&scalar);
+pub(crate) fn mul_affine_generic<Group: SWCurveConfig>(
+	base: Vec<u8>,
+	scalar: Vec<u8>,
+) -> Result<Vec<u8>, ()> {
+	let base = <ArkScale<short_weierstrass::Affine<Group>> as Decode>::decode(
+		&mut base.clone().as_slice(),
+	)
+	.map_err(|_| ())?;
+	let scalar =
+		<ArkScale<Vec<u64>> as Decode>::decode(&mut scalar.clone().as_slice()).map_err(|_| ())?;
```

Comment:
We'll remove this and make `mul_affine` just call `mul_projective` in ark-substrate, but I guess that's a later PR.